website/docs/r/dataflow_job.html.markdown

```diff
@@ -0,0 +1,56 @@
+---
+layout: "google"
+page_title: "Google: google_dataflow_job"
+sidebar_current: "docs-google-dataflow-job"
+description: |-
+  Creates a job in Dataflow according to a provided config file.
+---
+
+# google\_dataflow\_job
+
+Creates a job on Dataflow, which is an implementation of Apache Beam running on Google Compute Engine. For more information see
+the official documentation for
+[Beam](https://beam.apache.org) and [Dataflow](https://cloud.google.com/dataflow/).
+
+
+## Example Usage
+
+```hcl
+resource "google_dataflow_job" "big_data_job" {
+    name = "dataflow-job"
+	template_gcs_path = "gs://my-bucket/templates/template_file"
+    temp_gcs_location = "gs://my-bucket/tmp_dir"
+    parameters {
+        foo = "bar"
+        baz = "qux"
+    }
+}
+```
+
+## Note on "destroy" / "apply"
+There are many types of Dataflow jobs.  Some Dataflow jobs run constantly, getting new data from (e.g.) a GCS bucket, and outputting data continuously.  Some jobs process a set amount of data then terminate.  All jobs can fail while running due to programming errors or other issues.  In this way, Dataflow jobs are different from most other Terraform / Google resources.
+
+The Dataflow resource is considered 'existing' while it is in a nonterminal state.  If it reaches a terminal state (e.g. 'FAILED', 'COMPLETE', 'CANCELLED'), it will be recreated on the next 'apply'.  This is as expected for jobs which run continously, but may surprise users who use this resource for other kinds of Dataflow jobs.
+
+A Dataflow job which is 'destroyed' may be "cancelled" or "drained".  If "cancelled", the job terminates - any data written remains where it is, but no new data will be processed.  If "drained", no new data will enter the pipeline, but any data currently in the pipeline will finish being processed.  The default is "cancelled", but if a user sets `on_delete` to `"drain"` in the configuration, you may experience a long wait for your `terraform destroy` to complete.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name for the resource, required by Dataflow.
+    Changing this forces a new resource to be created.
```

Comment:
I would remove this line since changing any of the fields forces a new resource to be created.

Reply:
Done.